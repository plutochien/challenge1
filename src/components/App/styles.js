import styled from 'styled-components';

export const Layout = styled.div`
  display: flex;
  height: calc(100vh - 100px);
  width: calc(100vw - 100px);
  flex-direction: column;
  align-items: center;
  padding: 50px;
  background: white;
`;

export const Container = styled.section`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  min-height: 500px;
  max-width: 1000px;
  background: #d1ecf1;
  border-radius: 25px;
`;

export const Header = styled.header`
  display: flex;
  padding: 16px 16px 16px 16px;
  border: none;
  background: rgba(0, 0, 0, 0.003);
  box-shadow: inset 0 -2px 1px rgba(0, 0, 0, 0.03); ;
`;
export const Main = styled.div`
  min-height: 400px;
  overflow: auto;
`;

export const Footer = styled.footer`
  display: flex;
  margin-top: auto;
  padding: 10px 15px;
  text-align: center;
`;

export const Text = styled.span`
  flex: 1 1;
  color: #0c5460;
  text-align: justify;
  word-break: break-all;
  font-size: 20px;
  font-weight: 300;
  word-break: normal;
`;
