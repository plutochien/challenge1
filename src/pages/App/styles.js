import styled from 'styled-components';

export const Layout = styled.div`
  display: flex;
  padding: 0 400px;
  font: 16px "Roboto", sans-serif;
  @media (max-width: 1915px) {
    padding: 0 300px;
  }
  @media (max-width: 1432px) {
    padding: 0 200px;
  }
  @media (max-width: 1232px) {
    padding: 0 100px;
  }
  @media (max-width: 1032px) {
    padding: 0;
  }
`;

export const Container = styled.section`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  border-left: 2px solid rgb(231, 237, 241);
  border-right: 2px solid rgb(231, 237, 241);
  width:90%;
`;
