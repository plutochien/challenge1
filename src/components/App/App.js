import React, { Fragment } from 'react';

import AddTask from '../AddTask/AddTask';
import Filters from '../Filters/Filters';
import TodoList from '../TodoList/TodoList';
import { Container, Footer, Header, Layout, Main, Text } from './styles';

const App = () => (
  <Fragment>
    <Layout>
      <Container>
        <Header>
          <Text>{new Date().toJSON().slice(0, 10)}</Text>
          <Filters />
        </Header>
        <Main>
          <TodoList />
        </Main>
        <Footer>
          <AddTask />
        </Footer>
      </Container>
    </Layout>
  </Fragment>
);

export default App;
