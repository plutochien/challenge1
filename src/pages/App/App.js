import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Container, Layout } from './styles';
import Navigation from '../../components/Navigation/Navigation';
import Routes from '../../routes';

const App = () => (
  <Layout>
    <BrowserRouter>
      <Navigation />
      <Container>
        <Routes />
      </Container>
    </BrowserRouter>
  </Layout>
);

export default App;
