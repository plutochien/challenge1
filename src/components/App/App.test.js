import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import App from './App';
const mockStore = configureStore([]);

describe('AddTask Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      todos: {
        list: [
          {
            id: 7,
            taskText: 'task1',
            isComplete: true,
          },
          {
            id: 6,
            taskText: 'task2',
            isComplete: false,
          },
          {
            id: 1,
            taskText: 'task3',
            isComplete: true,
          },
        ],
      },
      filter: 'SHOW_ALL',
    });
    store.dispatch = jest.fn();
    component = renderer.create(
        <Provider store={store}>
          <App/>
        </Provider>,
    );
  });

  it.skip('should render with given state from Redux store', () => {
    jest.mock('react-beautiful-dnd', () => ({
      Droppable: ({ children }) => children({
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      }, {}),
      Draggable: ({ children }) => children({
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      }, {}),
      DragDropContext: ({ children }) => children,
    }));
    expect(component.toJSON()).toMatchSnapshot();
  });
})