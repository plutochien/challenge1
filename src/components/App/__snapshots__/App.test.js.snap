// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render with given state from Redux store 1`] = `
<div
  className="sc-kEjbxe bToKLA"
>
  <section
    className="sc-iqHYGH zbaCZ"
  >
    <header
      className="sc-crrsfI haQAoc"
    >
      <span
        className="sc-kstrdz VPfOD"
      >
        2020-11-29
      </span>
      <div>
        <button
          className="sc-eCssSg laEdmT"
          onClick={[Function]}
        >
          All
        </button>
        <button
          className="sc-eCssSg kUyIdX"
          onClick={[Function]}
        >
          Complete
        </button>
        <button
          className="sc-eCssSg kUyIdX"
          onClick={[Function]}
        >
          Incomplete
        </button>
      </div>
    </header>
    <div
      className="sc-dQppl fDyFKA"
    >
      <div>
        <li
          className="sc-jSgupP hEqZpK"
        >
          <span
            className="sc-iBPRYJ ctnspg"
          >
            1
             
          </span>
          <span
            className="sc-gKsewC bKHkhV"
          >
            task1
            <span
              className="sc-jrAGrp kqSLPU"
            >
              Done
            </span>
          </span>
          <button
            className="sc-fubCfw ewdZni"
            onClick={[Function]}
            size="40"
          >
            Undone
          </button>
          <button
            className="sc-pFZIQ egalCV"
            onClick={[Function]}
            size="40"
          >
            Delete
          </button>
        </li>
      </div>
      <div>
        <li
          className="sc-jSgupP hEqZpK"
        >
          <span
            className="sc-iBPRYJ evAWQB"
          >
            2
             
          </span>
          <span
            className="sc-gKsewC csViYh"
          >
            task2
            
          </span>
          <button
            className="sc-fubCfw ercPJq"
            onClick={[Function]}
            size="40"
          >
            Done
          </button>
          <button
            className="sc-pFZIQ egalCV"
            onClick={[Function]}
            size="40"
          >
            Delete
          </button>
        </li>
      </div>
      <div>
        <li
          className="sc-jSgupP hEqZpK"
        >
          <span
            className="sc-iBPRYJ cqYRLA"
          >
            3
             
          </span>
          <span
            className="sc-gKsewC crzOrt"
          >
            task3
            <span
              className="sc-jrAGrp kqSLPU"
            >
              Done
            </span>
          </span>
          <button
            className="sc-fubCfw ewdZni"
            onClick={[Function]}
            size="40"
          >
            Undone
          </button>
          <button
            className="sc-pFZIQ egalCV"
            onClick={[Function]}
            size="40"
          >
            Delete
          </button>
        </li>
      </div>
    </div>
    <footer
      className="sc-bqyKva dtreEc"
    >
      <input
        className="sc-bdfBwQ dQhNnk"
        placeholder="Add task"
      />
      <div
        className="sc-dlfnbm ggRcYP"
      >
        <input
          checked={false}
          className="sc-bdfBwQ eywrwi"
          name="priority"
          onChange={[Function]}
          type="radio"
          value="0"
        />
        <label
          className="sc-hKgILt bbXJBi"
        >
          Urgent
        </label>
        <input
          checked={true}
          className="sc-bdfBwQ eywrwi"
          name="priority"
          onChange={[Function]}
          type="radio"
          value="1"
        />
        <label
          className="sc-hKgILt hQUimg"
        >
          Medium
        </label>
        <input
          checked={false}
          className="sc-bdfBwQ eywrwi"
          name="priority"
          onChange={[Function]}
          type="radio"
          value="2"
        />
        <label
          className="sc-hKgILt kHPCfL"
        >
          Low
        </label>
      </div>
      <button
        className="sc-gsTCUz iWLRvp"
        onClick={[Function]}
        type="submit"
      >
        Add
      </button>
    </footer>
  </section>
</div>
`;
